sycl/include/syclcompat/math.hpp

```diff
@@ -47,6 +47,34 @@ namespace complex_namespace = sycl::ext::oneapi::experimental;
 template <typename ValueT>
 using complex_type = detail::complex_namespace::complex<ValueT>;
 
+template <typename ValueT>
+inline ValueT clamp(ValueT val, ValueT min_val, ValueT max_val) {
+  return sycl::clamp(val, min_val, max_val);
+}
+
+#ifdef SYCL_EXT_ONEAPI_BFLOAT16_MATH_FUNCTIONS
+// TODO: Follow the process to add this to the extension. If added,
+// remove this functionality from the header.
+template <>
+inline sycl::ext::oneapi::bfloat16 clamp(sycl::ext::oneapi::bfloat16 val,
+                                         sycl::ext::oneapi::bfloat16 min_val,
+                                         sycl::ext::oneapi::bfloat16 max_val) {
+  if (val < min_val)
+    return min_val;
+  if (val > max_val)
+    return max_val;
+  return val;
+}
+#endif
+
+template <typename ValueT>
+inline sycl::marray<ValueT, 2> clamp(sycl::marray<ValueT, 2> val,
+                                     sycl::marray<ValueT, 2> min_val,
+                                     sycl::marray<ValueT, 2> max_val) {
+  return {clamp(val[0], min_val[0], max_val[0]),
+          clamp(val[1], min_val[1], max_val[1])};
+}
```

Comment:
We don't have the `sycl::vec` version because these are not exposed in the `syclcompat` namespace (they live within `syclcompat::detail`)

However, your question raises a different issue. This code is not really used at all within syclcompat. I think It's better to remove it until an exposed API makes use of it.

Reply:
Makes sense, I think it is best removed if does not get used by anything